great_expectations/checkpoint/actions.py

```diff
@@ -815,7 +815,8 @@ def _run(
 
         checkpoint_ge_cloud_id = None
         if self.data_context.ge_cloud_mode and checkpoint_identifier:
-            checkpoint_ge_cloud_id = checkpoint_identifier.ge_cloud_id
+            checkpoint_ge_cloud_id = checkpoint_identifier if isinstance(checkpoint_identifier, str) else \
+                checkpoint_identifier.ge_cloud_id
```

Comment:
This patch was to address an exception that was being thrown only when updating an existing Checkpoint, using `context.add_checkpoint` (see update checkpoint notebook cell).

For some reason, `checkpoint_identifier` was coming back as a `str` only when updating. NB - I didn't actually confirm that the `str` `checkpoint_identifier` was actually the `ge_cloud_id`.

Reply:
is this change necessary? I would expect `add_checkpoint` to fail completely if the checkpoint already exists. We should use `save_checkpoint` for existing checkpoints